File: app/src/main/java/com/example/newyorktimearticles/data/webservice/ApiService.java
package com.example.newyorktimearticles.data.webservice;

import com.example.newyorktimearticles.data.model.Result;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiService {
    @GET("mostpopular/v2/viewed/7.json")
    Single<Result> getPopularArticles(@Query("api-key") String ApiKey);
}

File: app/src/main/java/com/gemidroid/newyorktimearticles/data/webservice/ApiService.kt
package com.gemidroid.newyorktimearticles.data.webservice

import com.gemidroid.newyorktimearticles.data.model.Result
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiService {
    @GET("mostpopular/v2/viewed/7.json")
    suspend fun getPopularArticles(@Query("api-key") ApiKey: String): Result
}